src/coins.h

```diff
@@ -130,6 +157,56 @@ struct CCoinsCacheEntry
     CCoinsCacheEntry() : flags(0) {}
     explicit CCoinsCacheEntry(Coin&& coin_) : coin(std::move(coin_)), flags(0) {}
     CCoinsCacheEntry(Coin&& coin_, unsigned char flag) : coin(std::move(coin_)), flags(flag) {}
+    ~CCoinsCacheEntry()
+    {
+        // We must always clear the flags when destroying this to remove it from
+        // the flagged linked list
```

Comment:
Is this really obvious though that we need to clear flags before destruction? Normally flags wouldn't affect destroying an object, but in this case they also tie the object to a linked list.

Reply:
No, but it's in a destructor, calling the `ClearFlags` method - I didn't seen what the comment added beyond what the code already explained well.
It could explain why you chose this style, or something that the code cannot, but this just seemed redundant to me.